ext/js/language/translator.js

```diff
@@ -636,10 +637,11 @@ export class Translator {
      * @param {string} text
      * @returns {string}
      */
-    _getJapaneseOnlyText(text) {
+    _getJapaneseChineseOnlyText(text) {
         let length = 0;
         for (const c of text) {
-            if (!isCodePointJapanese(/** @type {number} */ (c.codePointAt(0)))) {
+            const codePoint = /** @type {number} */ (c.codePointAt(0));
+            if (!isCodePointJapanese(codePoint) && !isCodePointChinese(codePoint)) {
```

Comment:
Considering we have this comment
https://github.com/themoeway/yomitan/blob/master/ext/js/language/ja/japanese.js#L41
I assume this is a hot code path and calling both in sequence should be avoided (a lot of the CJK stuff are in common), maybe making a set? Somehow

Reply:
Considering how this check functions I dont think this is an issue. It isnt running a check through every codepoint it's just checking if an integer is contained within two integers.

```js
export function isCodePointInRanges(codePoint, ranges) {
    for (const [min, max] of ranges) {
        if (codePoint >= min && codePoint <= max) {
            return true;
        }
    }
    return false;
}
```

At minimum to check both japanese and chinese this will have to run 40 times. The slightly suboptimal way I have it here makes it run 52 times. Integer equalities are way way way too fast to be worried about this imo. Especially since fixing this is quite tricky without making a mess or setting up a future mess for someone else.